### PR TITLE
Replace RestSharp with HttpClient.

### PR DESCRIPTION
**Pull request by rwecho**
Fixes ##313 #411 
Hi, there is the same issue in my project. I can not find a better workaround. So I  replace restsharp with HTTP client.  I have passed some tests.

Hope there's some help.
best regards.

### CODE REVIEW

**Comment by yilezhu** [comment]
Hi Friends, can you show me your code migration, I tried to migrate, but failed! Because the change is too big, you still need to read the restharp source code!

**Comment by rwecho** [comment]
@yilezhu 
The biggest change is MinioClient/Bucket/Object/V4Authenticator .
I wrap requests with HttpRequestMessageBuilder , and the response with ResponseResult.
you can find [details ](https://github.com/rwecho/minio-dotnet/commit/72c05b28128753007a2954f294e4de984e6eb4c0)

**Comment by poornas** [comment]
> I have passed some tests.
>
@rwecho, thanks for your contribution! did any tests in particular fail? With .netcore the functional tests pass. Have you tested with .Net framework also?

**Comment by rwecho** [comment]
> > I have passed some tests.
> 
> @rwecho, thanks for your contribution! did any tests in particular fail? With .netcore the functional tests pass. Have you tested with .Net framework also?


I run all of the build-in unit tests and run .netcore environment in my project. The point focused is TCP exhaustion, it got better.

File: Minio/ResponseResult.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (rwecho) [reply Minio/ResponseResult.cs:1]: ok

File: Minio/HttpRequestMessageBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+﻿using System;
```
[the remark this answers is not in the document]
Comment (rwecho) [reply Minio/HttpRequestMessageBuilder.cs:1]: ok

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 3]
```diff
@@ -2703,7 +2703,11 @@ internal async static Task ListenBucketNotificationsAsync_Test1(MinioClient mini
 
                         Assert.AreEqual(1, notification.Records.Length);
                         Assert.AreEqual("s3:ObjectCreated:Put", notification.Records[0].eventName);
-                        Assert.AreEqual(bucketName, notification.Records[0].s3.bucketMeta.name);
+                        if (notification.Records[0].s3.bucketMeta != null)
+                        {
+                            // todo s3 is null, how to complete this test.
```
[the remark this answers is not in the document]
Comment (rwecho) [reply Minio.Functional.Tests/FunctionalTest.cs:2708]: ok

**Comment by sergeyshaykhullin** [comment]
Any chance to merge? 👀

**Comment by poornas** [comment]
> Any chance to merge?
This is in progress. A lot of changes went into master, and a PR based off of this is being worked on.

**Pull request closed without merging**